mypyc/genops.py

```diff
@@ -1344,6 +1348,33 @@ def else_block() -> None:
 
         self.for_loop_helper(s.index, s.expr, body, else_block if s.else_body else None, s.line)
 
+    def spill_if_needed(self, value: Value, line: int, can_assign: bool = False,
+                        is_temp: bool = False) -> Union[Value, AssignmentTarget]:
```

Comment:
I would not bother with `is_temp`. Just always use a temp counter, or always base it on the name, or something, but I don't think it is worth cluttering up the callers with `is_temp=True`.

Perhaps `maybe_spill` is a nicer name? I dunno.

Also I now think that my `Union[Value, AssignmentTarget]` suggestion wasn't a great one, since I somehow didn't realize that some would need to actually be assigned to, which means that `asserts` are needed if this returns a `Union` (this is the typical downside of returning `Union`s). So I'd say either have a separate function for when it is assignable or just add a new sort of `AssignmentTarget`.